salt/lib.sls

```diff
@@ -261,3 +263,34 @@ add .htpasswd-{{ name }}-{{ username }}:
     - watch_in:
       - module: nginx-reload
 {% endmacro %}
+
+
+{% macro aws_site_backup(userdir, backup_location) %}
+{{ set_config('aws-settings.local', 'S3_SITE_BACKUP_BUCKET', backup_location) }}
+
+/home/sysadmin-tools/bin/site-backup-to-s3.sh:
+  file.managed:
+    - source: salt://files/site-backup-to-s3.sh
+    - mode: 750
+    - require:
+      - file: /home/sysadmin-tools/bin
+      - sls: aws
+
+/etc/cron.d/site_file_backup:
+  file.managed:
+    - contents: |
+        MAILTO=root
+        15 04 * * * root /home/sysadmin-tools/bin/site-backup-to-s3.sh
+    - require:
+      - file: /home/sysadmin-tools/bin/site-backup-to-s3.sh
+
+set BACKUP_DIRECTORIES setting:
+  file.keyvalue:
+    - name: /home/sysadmin-tools/aws-settings.local
+    - key: BACKUP_DIRECTORIES
+    - value: ( "{{ userdir }}/public_html/" )
+    - append_if_not_found: True
+    - require:
+      - file: /home/sysadmin-tools/bin
+      - sls: aws
```

Comment:
The only difference is that the previous code has `require`, `- sls: aws` but this isn't necessary.
Agreed and changed in new commit.

Reply:
I have found why this was separate, it is a very slight difference but the value for `BACKUP_DIRECTORIES` is surrounded by parentheses, rather than quotes. This is essential for bash to interpret the value as an array rather than string.
I would like to retain support for multiple backup locations and have modified the macro to better support this.